src/Config.php

```diff
@@ -73,10 +80,21 @@ class Config
      */
     public function __construct(string $path, bool $fileExists = false, array $settings = [])
     {
+        $pluginSettings = $settings['plugins'] ?? [];
+        unset($settings['plugins']);
```

Comment:
From my point it will be clean up PHP itself should be unscary

Reply:
I'm not sure what you're saying.

Reply:
I think Lars is referring to the `unset` and that PHPs internal garbage collector will clean it up automatically. But I'm not sure either :)

Reply:
Oh, I'm unsetting it here so that it doesn't cause any problems in `getJsonArray()` here:
https://github.com/captainhookphp/captainhook/pull/125/files#diff-9727fe62ad60c6637a7def31b06f908a39f99f118a78280469d7a62eb67a9d68R260-R267

Maybe this is unnecessary?

Reply:
Nope, totally necessary since you are not redefining `$config['plugins']` you are just appending.

```php
$data['config']['plugins'][] = $plugin->toJsonData();
```

Maybe this is a good hint to change it up, remove the unset, but completely overwrite `$config['plugins']` in `toJsonData` to be sure.